tests/test_loss_auto.py

```diff
@@ -214,7 +219,7 @@ def test_multiclass_single_output_unit():
 def test_binary_multiple_output_units():
```

Comment:
Can't binary classification work with loss="categorical_crossentropy"? If so, shouldn't that information be conveyed in the error message?

Reply:
This should be clearly mentioned on the documentation. I think that's appropriate for this PR.

Reply:
Here is the new section in the docs: https://www.adriangb.com/scikeras/refs/pull/218/merge/advanced.html#loss-selection